initramfs/init.sh

```diff
@@ -346,9 +342,24 @@ find_and_run_update()
             echo "Warning: unable to unmount '${UPDATE_IMG_MOUNT}'."
         fi
 
-        echo "Got '${SYSTEM_UPDATE_ENTRYPOINT}' script, trying to execute."
-        if ! "${update_tmpfs_mount}/${SYSTEM_UPDATE_ENTRYPOINT}" "${update_tmpfs_mount}/${UPDATE_IMAGE}" "${EMMC_DEV}" "${DISPLAY_TYPE}" "${SOFTWARE_INSTALL_MODE}"; then
-            echo "Error, update failed: executing '${update_tmpfs_mount}/${SYSTEM_UPDATE_ENTRYPOINT} ${update_tmpfs_mount}/${UPDATE_IMAGE} ${EMMC_DEV} ${DISPLAY_TYPE} ${SOFTWARE_INSTALL_MODE}'."
+
+        # Read the article number from the I2C device
+        art_num=$(i2ctransfer -y 1 w2@0x57 0x01 0x00 r4)
+        echo "---> Article number read from EEPROM: >${art_num}<"
+
+        # Convert the hexadecimal article number to a single decimal number
+        # Assuming art_num is in the format "0x00 0x03 0x78 0x34"
+        art_num_hex=""
+        for hex in $art_num; do
+            art_num_hex="${art_num_hex}${hex#0x}"  # Remove the "0x" prefix
+        done
+        art_num_dec=$(printf "%d\n" "0x$art_num_hex")
+
+        echo "---> Article number in decimal: >${art_num_dec}<"
```

Comment:
We already do this procedure in the function `check_and_set_eeprom_data`.
Move the call to `check_and_set_eeprom_data` before this function and set a BOM_NUMBER variable with it (swap lines 473 and 474). 
That will also fix the case of a first boot, where we still don't have an article number in the EEPROM, but have the USB Drive with it.